### PR TITLE
Add jvm-jersey library

### PR DESCRIPTION
**Pull request by sahil-lakhwani**
Signed-off-by: sahil-lakhwani <sahilakhwani@gmail.com>

This PR adds a new package for functions to work with JVM-Jersey environments.

It also modifies the current package name from `fission-java-core` to `fission-java-spring` to avoid any confusions.

### CODE REVIEW

File: fission-java-jersey/pom.xml [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+    </license>
+  </licenses>
+  <scm>
+    <url>https://github.com/fission/fission-java-libs.git</url>
+  </scm>
+  <developers>
+    <developer>
+      <name>Fission team</name>
+      <organization>Platform9</organization>
+      <organizationUrl>https://platform9.com/</organizationUrl>
```
Comment (vishal-biyani) [inline fission-java-jersey/pom.xml]: Please remove this reference - instead add InfraCloud reference

File: fission-java-spring/pom.xml [hunk 2]
```diff
@@ -2,7 +2,7 @@
   <modelVersion>4.0.0</modelVersion>
   
   <groupId>io.fission</groupId>
-  <artifactId>fission-java-core</artifactId>
+  <artifactId>fission-java-spring</artifactId>
```
Comment (vishal-biyani) [inline fission-java-spring/pom.xml]: In order to support backward compatibility - we should let `fission-java-core` be as it is

**Review by vishal-biyani — CHANGES_REQUESTED** [review]
Apart from the issues around backward compatibility - which means we should not rename `fission-java-core` to `fission-java-spring` - anywhere I think this PR is fine. I will approve it once I have tried the jersey environment.

**Review by vishal-biyani — CHANGES_REQUESTED** [review]
Hey @sahil-lakhwani there is some clean up needed:
1. FissionContext from java-core has been deleted (Need not be)
2. .gitignore moved out from java-core (need to be in both places)
3. I still see some fission-java-spring folder
4. Same with Function - we don't need to delete the old one but one change does that.

Can we please get this fixed?

**Comment by sahil-lakhwani** [comment]
> Hey @sahil-lakhwani there is some clean up needed:
> 
>     1. FissionContext from java-core has been deleted (Need not be)
> 
>     2. .gitignore moved out from java-core (need to be in both places)
> 
>     3. I still see some fission-java-spring folder
> 
>     4. Same with Function - we don't need to delete the old one but one change does that.
> 
> 
> Can we please get this fixed?

1 & 4 are done to keep the class and interface names in sync with our artifacts

I have changed the folder name to `fission-java-core`, as it was
Also, the new environment artifact is named `fission-jvm-jersey` with version `0.0.1`

**Comment by vishal-biyani** [comment]
Thanks @sahil-lakhwani this looks good!

**Pull request merged**